core/templates/dev/head/components/button-directives/create-activity-button.directive.ts

```diff
@@ -120,18 +107,31 @@ angular.module('oppia').directive('createActivityButton', [
               });
             }
           };
+          ctrl.$onInit = function() {
+            ctrl.creationInProgress = false;
+            ctrl.allowYamlFileUpload = ALLOW_YAML_FILE_UPLOAD;
 
-          // If the user clicked on a 'create' button to get to the dashboard,
-          // open the create modal immediately (or redirect to the exploration
-          // editor if the create modal does not need to be shown).
-          if (UrlService.getUrlParams().mode === 'create') {
-            if (!ctrl.canCreateCollections) {
-              ExplorationCreationService.createNewExploration();
-            } else {
-              ctrl.initCreationProcess();
+            ctrl.canCreateCollections = null;
+            ctrl.userIsLoggedIn = null;
+            UserService.getUserInfoAsync().then(function(userInfo) {
+              ctrl.canCreateCollections = userInfo.canCreateCollections();
+              ctrl.userIsLoggedIn = userInfo.isLoggedIn();
+            });
+
+            ctrl.showUploadExplorationModal = (
```

Comment:
It is better to have a new function constructed everywhere since it will create issues when services are migrated and functions are bound with this. So, I have updated assignments to new functions everywhere.